opm/autodiff/FlowMainEbos.hpp

```diff
@@ -86,21 +87,174 @@ namespace Opm
 
             Base::deck_ = ebosSimulator_->gridManager().deck();
             Base::eclipse_state_ = ebosSimulator_->gridManager().eclState();
-            IOConfig& ioConfig = Base::eclipse_state_->getIOConfig();
-            ioConfig.setOutputDir(Base::output_dir_);
+
+            try {
+                if (Base::output_cout_) {
+                    MissingFeatures::checkKeywords(*Base::deck_);
+                }
+
+                IOConfig& ioConfig = Base::eclipse_state_->getIOConfig();
+                ioConfig.setOutputDir(Base::output_dir_);
+
+                // Possible to force initialization only behavior (NOSIM).
+                if (Base::param_.has("nosim")) {
+                    const bool nosim = Base::param_.get<bool>("nosim");
+                    ioConfig.overrideNOSIM( nosim );
+                }
+            }
+            catch (const std::invalid_argument& e) {
+                std::cerr << "Failed to create valid EclipseState object. See logfile: " << logFile_ << std::endl;
+                std::cerr << "Exception caught: " << e.what() << std::endl;
+                throw;
+            }
 
             // Possibly override IOConfig setting (from deck) for how often RESTART files should get written to disk (every N report step)
             if (Base::param_.has("output_interval")) {
                 const int output_interval = Base::param_.get<int>("output_interval");
                 eclipse_state_->getRestartConfig().overrideRestartWriteInterval( size_t( output_interval ) );
 
             }
+        }
+
+        // Create grid and property objects.
+        // Writes to:
+        //   grid_init_
+        //   material_law_manager_
+        //   fluidprops_
+        //   rock_comp_
+        //   gravity_
+        //   use_local_perm_
+        //   geoprops_
+        void setupGridAndProps()
+        {
+            Dune::CpGrid& grid = ebosSimulator_->gridManager().grid();
+
+            //Base::material_law_manager_ = ebosSimulator_->problem().materialLawManager();
+
+            grid.switchToGlobalView();
```

Comment:
I guess that this statement a no op-here: it is already done when the ebos simulator creates its grid (thus the distinction between equilGrid() and grid().)